Classes/Service/BigQueryService.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace t3n\FlowLog\Service;
+
+use Google\Cloud\BigQuery\BigQueryClient;
+use Google\Cloud\BigQuery\Dataset;
+use Google\Cloud\BigQuery\Table;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Exception;
+
+/**
+ * @Flow\Scope("singleton")
+ */
+class BigQueryService
+{
+    /**
+     * @Flow\InjectConfiguration(package="t3n.FlowLog", path="bigQuery")
+     *
+     * @var mixed[]
+     */
+    protected $settings;
+
+    /**
+     * @var BigQueryClient
+     */
+    protected $bigQueryClient;
+
+    public function getClient(): BigQueryClient
+    {
+        if ($this->settings['keyFilePath'] === null || $this->settings['dataset'] === null || $this->settings['table'] === null) {
+            throw new Exception('BigQueryLogger is not configured.');
+        }
+
+        if (! $this->bigQueryClient instanceof BigQueryClient) {
+            $this->bigQueryClient = new BigQueryClient(['keyFilePath' => $this->settings['keyFilePath']]);
+        }
+
+        return $this->bigQueryClient;
+    }
+
+    public function getTable(): Table
+    {
+        $dataset = $this->getDataset();
+        $tableId = $this->settings['table'];
+
+        if ($dataset->table($tableId)->exists()) {
+            return $dataset->table($tableId);
+        }
+
+        return $dataset->createTable($tableId, [
+            'schema' => $this->getTableSchema(),
+            'timePartitioning' => [
+                'type' => 'DAY',
+                'expirationMS' => '7776000000',
+                'field' => 'date'
+            ],
+        ]);
+    }
+
+    protected function getDataset(): Dataset
+    {
+        return $this->getClient()->dataset($this->settings['dataset']);
+    }
+
+    /**
+     * @return mixed[]
+     */
+    protected function getTableSchema(): array
```

Comment:
Move this to Settings.yaml to make it easy configurable?

Reply:
I have moved the table schema. Now everybody is able to extend the logging behaviour.